### PR TITLE
[#115] Add iTaigi

### PR DESCRIPTION
**Pull request by M157q**
ref: #115

### CODE REVIEW

**Comment by M157q** [comment]
If this is ok, we can merge and release version 2.1.0.

**Comment by wdv4758h** [comment]
@M157q 	I make it crash at first try :stuck_out_tongue: 

```
$ zdict -dt itaigi -d 測試
測試
  測試 tshik-tshì
Traceback (most recent call last):
  File "/home/user/py/venv/bin/zdict", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/user/zdict/scripts/zdict", line 4, in <module>
    zdict.main()
  File "/home/user/zdict/zdict/zdict.py", line 312, in main
    execute_zdict(args)
  File "/home/user/zdict/zdict/zdict.py", line 294, in execute_zdict
    normal_mode(args)
  File "/home/user/zdict/zdict/zdict.py", line 204, in normal_mode
    zdict.lookup(word)
  File "/home/user/zdict/zdict/dictionary.py", line 157, in lookup
    self.show(record)
  File "/home/user/zdict/zdict/dictionaries/itaigi.py", line 184, in show
    if content['related_words']:
KeyError: 'related_words'
```

**Comment by wdv4758h** [comment]
LGTM overall

File: README.rst [hunk 1]
```diff
@@ -203,6 +203,14 @@ To use this source, you should first `apply <https://developer.oxforddictionarie
 .. image:: https://i.imgur.com/UO5nQjU.png
 
 
+`iTaigi-愛台語 <https://itaigi.tw/>`_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. image:: https://user-images.githubusercontent.com/1645228/55276638-07bf6f00-5331-11e9-990c-648595295e29.jpg
```
Comment (iblis17) [inline README.rst]: There are some tofus in the screenshot. Is it possible to get rid of it?

File: zdict/dictionaries/itaigi.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+        try:
+            _ = json.loads(requests.get(url).text)
+        except Exception:
+            return {}
+
+        try:
+            mandarin_sentence = _["例句"][0]["華語"]
+        except Exception:
+            mandarin_sentence = None
```
Comment (iblis17) [inline zdict/dictionaries/itaigi.py]: For this kind of pattern, I suggest this:
```python
mandarin_sentence = None
with suppress(KeyError):  # from contexlib import suppress
    mandarin_sentence = _["例句"][0]["華語"]
```

File: zdict/dictionaries/itaigi.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+        # Not Found
+        if not response.get("列表"):
+            raise NotFoundError(word)
+
+        # Show Chinese word from iTaigi in stead of user input if possible
+        try:
+            word = response["列表"][0]["外語資料"]
+        except Exception:
+            pass
```
Comment (iblis17) [inline zdict/dictionaries/itaigi.py]: `suppress`

File: zdict/tests/dictionaries/test_itaigi.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+from pytest import raises
+from unittest.mock import Mock, patch
+
+from zdict.exceptions import NotFoundError
+from zdict.zdict import get_args
+
```
Comment (iblis17) [inline zdict/tests/dictionaries/test_itaigi.py]:
```suggestion
```

File: README.rst (hunk 1 above)
Comment (M157q) [reply README.rst]: sure. Let me find another font.

**Pull request merged**